### ISSUE TITLE
Unable to use Multiple Event Hub Output Bindings 

### ISSUE THREAD

**Issue by MarekLani** [Azure/azure-functions-java-worker#691]
I am developing Java Azure Function which is supposed to output messages into two different event hubs, based on certain decision logic. Nevertheless, due to some issue with bindings, messages always end up sent only to one of the event hubs. I see this problem during local development, haven't tested it on Azure.

I mapped two different output bindings in the following way:
```
    @FunctionName("events")
    public HttpResponseMessage run(
            @HttpTrigger(name = "req", methods = { HttpMethod.GET,
                    HttpMethod.POST }, authLevel = AuthorizationLevel.ANONYMOUS) HttpRequestMessage<Optional<String>> request,
            @EventHubOutput(name = "asaevent", eventHubName = "", connection = "ASAEventHubSendConnectionString") OutputBinding<List<EventData>> asaEventOutput,
            @EventHubOutput(name = "processorfuncevent", eventHubName = "", connection = "ProcessorFuncEventHubSendConnectionString") OutputBinding<List<EventData>> processorFuncEventOutput,
            final ExecutionContext context)
```

And I use them like this:

```
            // Determining which event hub to be targeted
            if (simpleEventTypes.contains(event.eventType)) {
                processorFuncEventOutput.setValue(events);
            } else {
                asaEventOutput.setValue(events);
            }
```

I should be seeing messages landing in both Event Hubs, for some reason, both output bindings end up pointing to just one of the event hubs. From observations it seems that first messages sort of decides, which event hub will be used for all messages and both outputs continue pointing to same even hub.

(I am sure connection strings are different and correct, because I tested it also with manually created Evet Hub Client and it worked, also I am sure both code paths in the condition above are activated)

#### Investigative information

Please provide the following:
 N/A as I am running it locally

- Timestamp:
Function name(s) (as appropriate):
- Invocation ID:
- Region:- Function App name:
- Function name(s) (as appropriate):
- Invocation ID:
- Region:

#### Repro steps

See above

#### Expected behavior

Output message lands in correct Event Hub

#### Actual behavior

Messages always land only in one of the event hubs mapped thru output biding

#### Known workarounds

Manually construct Event Hub messaging client and do not use Azure Function Bindings

```
    private static EventHubProducerClient producerProcessorEH = new EventHubClientBuilder()
            .connectionString(System.getenv("ProcessorFuncEventHubSendConnectionString"),
                    System.getenv("processorFuncEventHubName"))
            .buildProducerClient();

    private static EventHubProducerClient producerASAEH = new EventHubClientBuilder()
            .connectionString(System.getenv("ASAEventHubSendConnectionString"),
                    System.getenv("asaEventHubName"))
            .buildProducerClient();
```

#### Related information 

Language Used: Java 11
Core Tools Version:       4.0.4785 Commit hash: N/A  (64-bit)
Function Runtime Version: 4.10.4.19213

**Comment by shreyas-gopalakrishna** [comment]
@MarekLani Could you please share the function.json file generated when you run `mvn clean package`

**Comment by kamperiadis** [comment]
@MarekLani I was able to reproduce this issue consistently both locally and in production when I left the `eventHubName` parameter empty. However, it all behaved as expected when I set a value for `eventHubName`. Example:
```
 @FunctionName("HttpExample")
    public HttpResponseMessage run(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.GET, HttpMethod.POST},
                authLevel = AuthorizationLevel.ANONYMOUS)
                HttpRequestMessage<Optional<String>> request,
            @EventHubOutput(name="firstevent", eventHubName="eventhub1", connection="FirstEventHubConnectionString") OutputBinding<String> firstHub,
            @EventHubOutput(name="secondevent", eventHubName="eventhub2", connection="SecondEventHubConnectionString") OutputBinding<String> secondHub,
            final ExecutionContext context) 
```
Can you please try to see if doing this resolves the issue for you?

**Comment by MarekLani** [comment]
@shreyas-gopalakrishna before setting event hub names function.json file generated seems to be correct, but issue appears:
```
{
  "scriptFile" : "../Ingress_Function-1.0-SNAPSHOT.jar",
  "entryPoint" : "Omega5.function.ingress.IngestionHttpTriggerFunction.run",
  "bindings" : [ {
    "type" : "httpTrigger",
    "direction" : "in",
    "name" : "req",
    "methods" : [ "GET", "POST" ],
    "authLevel" : "ANONYMOUS"
  }, {
    "type" : "eventHub",
    "direction" : "out",
    "name" : "asaevent",
    "connection" : "ASAEventHubSendConnectionString",
    "eventHubName" : ""
  }, {
    "type" : "eventHub",
    "direction" : "out",
    "name" : "processorfuncevent",
    "connection" : "ProcessorFuncEventHubSendConnectionString",
    "eventHubName" : ""
  }, {
    "type" : "http",
    "direction" : "out",
    "name" : "$return"
  } ]
}
```

**Comment by MarekLani** [comment]
@kamperiadis, thank you I confirm, it works as desired when event hub name is stated explicitly. Nevertheless, I believe this should be still considered as an issue to be solved, because event hub name parameter should be optional if entity path is captured in connections string.

**Comment by kaibocai** [comment]
> @kamperiadis, thank you I confirm, it works as desired when event hub name is stated explicitly. Nevertheless, I believe this should be still considered as an issue to be solved, because event hub name parameter should be optional if entity path is captured in connections string.

@MarekLani can you share where did you see that event hub name is option with connections string present, we may need to re-check it's accuracy. Thanks.

**Comment by MarekLani** [comment]
Sorry, my bad, I used to work on c# functions extensively in the past and I haven't faced the issue there. I was using connection string with entity path, nevertheless now when reviewing docs it explicitly states: 

_Obtain this connection string by clicking the Connection Information button for the [namespace](https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-create#create-an-event-hubs-namespace), not the event hub itself. The connection string must be for an Event Hubs namespace, not the event hub itself.__

Closing this issue, thank you for guidance!
